src/backend_cypressfx3/sw/backend_cypressfx3.c

```diff
@@ -454,6 +455,26 @@ int gb_cypressfx3_close(struct glip_ctx *ctx)
         return -1;
     }
 
+    /* ensure the write buffer is empty before closing the connection */
+    struct glip_backend_ctx* bctx = ctx->backend_ctx;
+    size_t write_buf_fill_level = cbuf_fill_level(bctx->write_buf);
+    if (write_buf_fill_level > 0) {
+		struct timespec ts;
+		/* get current time */
+		timespec_get(&ts, TIME_UTC);
```

Comment:
`timespec_get()` is C11. In other code, we use the POSIX variant of that. Please, either switch occurrences to the C11-variant as well and specify C11 as compile option, or use the POSIX variant:

~~~
clock_gettime(CLOCK_REALTIME, &ts);
timespec_add_ns(&ts, timeout * 1000 * 1000);
~~~

timespec_add_ns is required to properly handle overflows into the second field.